### DISCUSSION TITLE
Error while downloading s3 file : operation error S3: GetObject, https response error StatusCode: 404

### DISCUSSION THREAD

**Discussion by suhasbansude** [aws/aws-sdk-go-v2#1940] [General]
! - _ . * ' ( )_test042022.98765432145698745.pdf

I am trying to download above file from s3 bucket. getting 404 error

downloader := manager.NewDownloader(awsS3Client)
	numBytes, err := downloader.Download(context.TODO(), newFile, &s3V2.GetObjectInput{
		Bucket: aws.String(APP_S3BUCKET_NAME),
		Key:    aws.String(s3FileKey),
	})

**Comment by suhasbansude** [comment]
Above issue if for * special character
But as per AWS document * is safe special character
https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html

**Comment by RanVaknin** [comment]
Hi @suhasbansud,

Since your code is missing, its impossible for me to know what the issue is. 

I've created a pdf and named it `! - _ . * ' ( )_test042022.98765432145698745.pdf` uploaded it to s3 and was able to successfully download it using this code:
As proof I printed the length of the buffer I created to write the downloaded object.

```go
func main() {
	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion("us-east-1"))
	if err != nil {
		log.Fatalf("unable to load SDK config, %v", err)
	}

	client := s3.NewFromConfig(cfg)

	downloader := manager.NewDownloader(client)

	var blob []byte
	buf := manager.NewWriteAtBuffer(blob)

        // buffer size before download
	fmt.Println("bytes: ", len(buf.Bytes()))

	_, err = downloader.Download(context.Background(), buf, &s3.GetObjectInput{
		Bucket: aws.String("foo-bucket"),
		Key:    aws.String("! - _ . * ' ( )_test042022.98765432145698745.pdf"),
	})
	if err != nil {
		panic(err)
	}

        // buffer size after download
	fmt.Println("bytes: ", len(buf.Bytes()))
}
```

Output:
```
bytes:  0
bytes:  465722
```
Let me know if this helps,
Thanks,
Ran
